gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/discord/Listener.java

```diff
@@ -57,9 +49,18 @@ public void onMessageReceived(@NotNull MessageReceivedEvent event) {
         // Check if the category is not null and then get its name
         String categoryName = (category != null) ? category.getName() : "No Category"; // You can change "No Category" to a default value if needed
 
+        // check if author is gendox user and if not, create new user
+        if (event.getAuthor().isBot()) return;
+        try {
+            if (!listenerService.IsUserExist(authorName)) {
+                listenerService.createDiscordUser(authorName);
+            }
+        } catch (GendoxException e) {
+            System.err.println("An An error occurred while checking/creating the user: " + e.getMessage());
+        }
```

Comment:
We should never swallow the exceptions. If this happen the request should fail. you can log the exception here with a **Logger** and throw it again.

Read:
https://www.baeldung.com/java-exceptions
https://www.harness.io/blog/swallowed-exceptions-java-applications